src/Extensions/Diagnostics.HealthChecks/Internal/HttpEndpointHealthCheck.cs

```diff
@@ -59,6 +60,11 @@ protected override async Task<HealthCheckResult> CheckHealthInternalAsync(Health
 
 			HttpRequestMessage request = new HttpRequestMessage(Parameters.Method, m_uriToCheck);
 
+			foreach(KeyValuePair<string, string> pair in Parameters.Headers)
+			{
+				request.Headers.Add(pair.Key, pair.Value);
```

Comment:
Do we want this to throw for invalid values?
If not, suggest using TryAddWithoutValidation